### PR TITLE
Feat(#54) Delete Rooms

### LINKED ISSUE #54: Delete rooms
### User story

As an administrator, I can delete a room and define its availabilities. 

### Narrative

Administrators must be able to delete rooms in the system. 

### Out of scope

Adding room capabilities (Projector, smart board, etc) 

### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am an adminstrator|I delete a new room|it is deleted from the sytem|
|2|I am not an administrator|I delete a new room|it is not deleted from the system|

### PR DESCRIPTION
**Pull request by Alex-Hayeur**
Adds a delete function on the list of all rooms + tests

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Adds the functionality for users to delete rooms via the list of all rooms.
Delete_Rooms_1
Delete_Rooms_2

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Tests ensure that removing room entries from the DB with the appropriate method results in the room no longer existing

### Closing issues
List all issues this PR resolves.
- Closes #54 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [ ] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [x] UI Implementation
- [ ] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [ ] Integration Tests
- [ ] System Tests

### CODE REVIEW

File: app/Providers/JetstreamServiceProvider.php [hunk 1]
```diff
@@ -25,7 +26,8 @@ public function register()
      */
     public function boot()
     {
-        $this->configurePermissions();
+        //$this->configurePermissions();
```
Comment (alexstojda) [inline app/Providers/JetstreamServiceProvider.php:29]: Was there a reason this needed to be commented out?

File: app/Providers/JetstreamServiceProvider.php [hunk 2]
```diff
@@ -46,4 +48,21 @@ protected function configurePermissions()
             'delete',
         ]);
     }
+    
+    /**
+     * Configure the routes offered by the application.
+     *
+     * @return void
+     */
+    protected function configureRoutes()
```
Comment (alexstojda) [inline app/Providers/JetstreamServiceProvider.php:57]: What is the reasoning for this change?

File: routes/inertia.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?php
```
Comment (alexstojda) [inline routes/inertia.php:1]: Same question here, not sure how this relates to your feature

Comment (Alex-Hayeur) [reply routes/inertia.php]: All 3 of these were me accidentally adding in some backend files from a different feature. I removed them in them most recent commit. Thanks for pointing them out.

**Review by alexstojda — CHANGES_REQUESTED** [review]
There seems to be something wrong with the php coverage reporting, but you are missing a test for your JS component. See the link below for the lines that need to be covered.

https://sonarcloud.io/component_measures?id=CSU-Booking-Platform_application&metric=new_coverage&pullRequest=207&view=list

**Comment by Alex-Hayeur** [comment]
> There seems to be something wrong with the php coverage reporting (i'm gonna take a look at that tonight), but you are missing a test for your JS component. See the link below for the lines that need to be covered.
> 
> https://sonarcloud.io/component_measures?id=CSU-Booking-Platform_application&metric=new_coverage&pullRequest=207&view=list

I've added the missing JS test, thanks for pointing that out. Hopefully that was what was causing the low code coverage %.

**Review by alexstojda — APPROVED** [review]
100% coverage! Woohoo!

**Review by jonathanp16 — APPROVED** [review]
Clean and simple. Good work.

**Pull request merged**